src/plugins/intel_cpu/src/plugin.h

```diff
@@ -56,6 +56,8 @@ class Engine : public InferenceEngine::IInferencePlugin {
        So track if streams is set explicitly (not auto-configured) */
     bool streamsExplicitlySetForEngine = false;
     const std::string deviceFullName;
+
+    std::shared_ptr<void> specialSetup;
```

Comment:
or unique_ptr ?

Reply:
`std::shared_ptr<T>` has following constructor which allows `std::shared_ptr<void>` to be constructed from any pointer of type Y with correct default deleter for type Y, `std::unique_ptr<T>`  don't have this constructor and caller have to explicitly specify the type of Deleter,  I prefer to not expose CPUSpecialSetup/SigAltStackSetup class to world outside of plugin.h and use `std::shared_ptr<void>` is more convinient
```
template< class Y >
explicit shared_ptr( Y* ptr );
```